pythonwhat/local.py

```diff
@@ -66,3 +53,46 @@ def run_code(process, code):
         raw_output = ""
         error = str(e)
     return raw_output, error
+
+
+def run_single_process(pec, code, pid=None):
+    process = StubProcess(init_code=pec, pid=pid)
+    raw_stu_output, error = run_code(process, code)
```

Comment:
If I understand correctly, this is just an `exec(code)`, right? So we could really mess things up from the student code, so the solution code doesn't run anymore?

I'm not saying we have to fix this, just want to make sure I understand.

Reply:
It execs with an empty environment (`exec(code, {})`) so it can't immediately interfere with the code.
Exceptions are caught in `run_code` (although the message could probably be improved to look like executing a real file, it's not shown to the user, only used in `has_no_error`).

Reply:
I also switched the order to first execute the solution, as that's also the order of the normal execution.

Reply:
After writing tests to see how it's not isolated, I improved this by using processes.